### DISCUSSION TITLE
"generate documentation template" code action

### DISCUSSION THREAD

**Discussion by bddap** [microsoft/pylance-release#2657] [Enhancement]
It would be nice to be able to generate a docstring based on the type signature. I think in general this can help encourage people to write docstrings.

Previous work:
- in rust analyzer: [generate_documentation_template](https://rust-analyzer.github.io/manual.html#generate_documentation_template). example: https://rust-analyzer.github.io/thisweek/2021/12/13/changelog-107.html#new-features
- in pycharm: https://www.jetbrains.com/help/pycharm/creating-documentation-comments.html#b1406e21

**Comment by bschnurr** [comment]
Thanks for the suggestion

**Comment by judej** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.
